homework/index.js

```diff
@@ -30,17 +34,212 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  // You can easily modify static html elements with this model
+  const componentModel = [
```

Comment:
While this is a creative alternative and original piece of work for defining the DOM structure, it is not necessarily better than the other approach, i.e. using direct calls to the DOM. To function as a virtual DOM you also need a 'diffing' algorithm which compares the virtual DOM to the real DOM and selectively updates the latter. If you want to learn more about how React does this, check this article:

https://medium.com/@gethylgeorge/how-virtual-dom-and-diffing-works-in-react-6fc805f9f84e

Reply:
I have reviewed the post about virtual dom. I understood the reason of virtual dom and diffing algorithm. I have also reviewed some posts about 'how to type your own virtual dom?'. But it's not requierement of the assignment and hard to make it for now.

Thanks for detailed informations.